app/lib/actions/pulp3/repository/save_version.rb

```diff
@@ -7,15 +7,24 @@ def plan(repository, tasks)
         end
 
         def run
-          version_href = input[:tasks].last[:created_resources].first
           repo = ::Katello::Repository.find(input[:repository_id])
-          repo_version = repo.backend_service(::SmartProxy.pulp_master).lookup_version version_href
+          version_href = input[:tasks].last[:created_resources].first
+
+          # Assumption that repo.version_href being nil means the repo was just created
+          if version_href.nil? && repo.version_href.nil?
+            version_href = input[:tasks].last[:reserved_resources_record].first + "versions/0/"
+          end
 
```

Comment:
Is there a valid case where repository is just created but version isn't created in the run phase of save_version?

Reply:
@sjha4 are you wondering if Pulp 3 could be late in making the first repo version? If so, it appears that the first version is created at the same time as repo creation, so I don't think it would ever be late.

Reply:
So when do you see this if condition being valid and needing to form the version href instead of getting it from pulp?

Reply:
oh, @sjha4 this condition is valid when the repo is first created.  `version_href` is nil because `input[:tasks].last[:created_resources].first` is now empty due to pulp 3 changes.  `repo.version_href` is nil because this is the first code that gets a chance to set its `version_href`.  I have to form the `version_href` because of how the version 0 creation happens silently with the repo creation.

Reply:
Maybe that's something pulp can return in created resources if we press on it..:smiley: